src/zsys.c

```diff
@@ -334,9 +334,7 @@ zsys_init (void)
         zsys_set_thread_name_prefix (s_thread_name_prefix);
 
     if (getenv ("ZSYS_THREAD_NAME_PREFIX_STR"))
-        zsys_set_thread_name_prefix_str (getenv ("ZSYS_THREAD_NAME_PREFIX"));
-    else
-        zsys_set_thread_name_prefix_str (s_thread_name_prefix_str);
+        zsys_set_thread_name_prefix_str (getenv ("ZSYS_THREAD_NAME_PREFIX_STR"));
```

Comment:
Ops the wrong env var is clearly a copypasta, my bad - but iirc the "else" branch is required. Why is it being removed?

Reply:
If you leave the else branch in, setting any setting of ZSYS_THREAD_NAME_PREFIX would be ignored. And it crashes because you can't strcpy(s,s) in a Mac (and it's illegal on Linux too, but apparently does not crash).

Reply:
Personally I think the interface with two env variables is confusing: if the user specifies both variables, which one takes precedence? Also, the underlying implementation in libzmq only uses a single variable to hold the config value, it just uses two different APIs for setting it as either an integer, or a string.

Reply:
I'd be happy to create a PR that would use only one env var and calls the correct api depending on whether it has to be a string or not, if you are willing to go this route.

Reply:
Ah yes I see the issue now - regarding the env var, it's just an artifact of how it was implemented in libzmq. It's now in a stable release, so it cannot be removed, but feel free to send a PR to change the implementation.

Reply:
Now I'm confused: I thought zsys_thread_name_prefix_str() and therefore also ZSYS_THREAD_NAME_PREFIX_STR is a draft API and draft APIs can be changed?

If we can't remove that env variable and only use ZSYS_THREAD_NAME_PREFIX as describe in PR comment, I don't know what the PR would contain.

Reply:
I was thinking about the old one - the new one is yes, so you can change that. But the behaviour of the old one cannot change - so existing uses cannot break.

Reply:
got it